### PR TITLE
function that returns a bool telling if a point belongs to the navmesh

### PR DESCRIPTION
**Pull request by RogerNogue**
Created a function that returns if a point belongs to the nav mesh.
To test this function a call with a point as a parameter is required

### CODE REVIEW

**Comment by BorjaSasieta** [comment]
can you specify better and give an example of testing?

File: Source/ModuleNavigation.cpp [hunk 1]
```diff
@@ -1434,6 +1434,24 @@ bool ModuleNavigation::NavigateTowardsCursor(math::float3 start, std::vector<mat
...
+	//parameters for the result and flags for the navmesh query
+	dtPolyRef navPoint;
+	dtQueryFilter filter;
+	filter.setIncludeFlags(SAMPLE_POLYFLAGS_ALL ^ SAMPLE_POLYFLAGS_DISABLED);
+	filter.setExcludeFlags(0);
+
+	//navmesh query
+	navQuery->findNearestPoly((float*)& point, polyPickExt, &filter, &navPoint, 0);
+
+	if (!navPoint)	return false;
```
Comment (BorjaSasieta) [inline Source/ModuleNavigation.cpp]: this line isn't correct!

use a key phase ({
})

**Comment by Rubetoman** [comment]
It would be cool to have a float2 version, so it can tell you if where you clicked is a navigable mesh. Also, some easier testing instructions would be better.

**Comment by AniolAndres** [comment]
> It would be cool to have a float2 version, so it can tell you if where you clicked is a navigable mesh. Also, some easier testing instructions would be better.

What's the point of having a float2 in a 3D mesh? Could you please elaborate?

**Pull request closed without merging**